docs/applications/configuration-management/configure-and-use-salt-ssh.md

```diff
@@ -0,0 +1,216 @@
+
+        $rpm -q salt
+        $rpm -q salt-ssh
+
+    {: .note}
+    >
+    > For detailed instruction on how to set up SaltStack repo, please refer to [SaltStack installation](https://www.linode.com/docs/applications/configuration-management/install-and-configure-salt-master-and-minion-servers)
+
+3.  Your minions must have Python installed. Without Python installed on minions, you will only be able to run salt-ssh in raw mode. If you're running any modern version of CentOS/RedHat, you should already have Python installed on your systems. In raw mode, you execute a raw shell command and cannot use execution modules or apply salt states.
+
+4.  You must have at least one master server and one minion (client).
+
+##Set up Salt Roster file
+
+Roster file contains information on target systems, connection details and credentials, which will be used for connection.
+Default location for roster file is `/etc/salt/roster`.
+
+   {: .note}
+   >
+   > Roster file is configured on master server.
+
+1.  Open the file `/etc/salt/roster` with your editor of choice and add following lines to define client systems:
+
+    This is an example of minimal host definition
+
+    {: .file }
+    /etc/salt/roster
+    :  ~~~ config
+    linode1:
+         host: <IPADDRESS OR HOSTNAME>
+         user: <username>
+         passwd: <password>
+       ~~~
+
+    {: .note}
+    >
+    > Roster file keeps data in yaml format so be careful with the amount of spaces you use.
+
+2.  To set up access to minion using private key. If you already have your public key installed on minion and you have private key on master, add following lines:
+    {: .file }
+    /etc/salt/roster
+    :  ~~~ config
+    #This is an example of minimal host definition using private key:
+    linode1:
+        host: <IPADDRESS OR HOSTNAME>
+        user: <username>
+        priv: /<username_home_folder>/.ssh/id_rsa
+       ~~~
+
+    {: .note}
+    >
+    > Using ssh keys is much more safer way to access your minions, as you can avoid storing password in plaintext.
+ 
+3.  To set up connection to minion as a regular user, you will have to add a few extra lines.In this case Salt will leverage privileges via sudo. Just set sudo value to True in host definition like in the example below. You may also have to add tty option if `/etc/sudoers` on your minion is configured with requiretty option:
+
+    {: .file-excerpt}
+    /etc/salt/roster
+    :  ~~~ config
+    linode1:
+        host: <IPADDRESS OR HOSTNAME>
+        user: <username>
+        passwd: <password>
+        sudo: True
+        tty: True
+       ~~~
+
+4.  Check that master server have access to client using salt-ssh command:
+
+        [root@master ~]# salt-ssh linode1 test.ping
+
+    You should see output similar to:
+
+        linode1:
+            True
+
+    {: .note}
+    >
+    > Permissions leverage via sudo works only if NOPASSWD option is set up for the user, which will be used to connect to the minion, in `/etc/sudoers`.
```

Comment:
Can you clarify on this point? I ended up using root credentials to install salt-minion via Salt SSH

```
linode2:
    ----------
    retcode:
        1
    stderr:
        sudo: no tty present and no askpass program specified
    stdout:
        ERROR: sudo expected a password, NOPASSWD required
```